diff --git a/PythonPackage.cmake b/PythonPackage.cmake
index 0f18a3b..e7dbcbc 100644
--- a/PythonPackage.cmake
+++ b/PythonPackage.cmake
@@ -623,6 +623,7 @@ function(add_setup_py target template)
         endforeach ()
 
         set(_lnk "")
+        message(STATUS "LNK: ${lnk}")
         foreach (item ${lnk})
             get_filename_component(_libdir ${item} DIRECTORY)
 
@@ -632,6 +633,11 @@ function(add_setup_py target template)
                 string(REGEX REPLACE "lib" "" item ${item})
             endif ()
 
+            if( ${item} MATCHES framework)
+                message(STATUS "::::::::::: FOUND FRAMEWORK ${item}")
+                set(item "-framework ${item}")
+            endif()
+
             list(APPEND _lnk "'${item}'")
         endforeach ()
 
