diff --git a/PythonPackage.cmake b/PythonPackage.cmake
index 0f18a3b..3aada8e 100644
--- a/PythonPackage.cmake
+++ b/PythonPackage.cmake
@@ -624,15 +624,18 @@ function(add_setup_py target template)
 
         set(_lnk "")
         foreach (item ${lnk})
+            message(STATUS "LNK ITEM: ${item}")
             get_filename_component(_libdir ${item} DIRECTORY)
+            get_filename_component(_name   ${item} NAME_WE)
 
-            if (_libdir)
+            if( ${item} MATCHES .framework)
+                list(APPEND _xlk "'-F ${_libdir}'")
+                list(APPEND _xlk "'-Wl,-framework ${_name}'")
+            elseif (_libdir)
                 list(APPEND _dir '${_libdir}')
-                get_filename_component(item ${item} NAME_WE)
-                string(REGEX REPLACE "lib" "" item ${item})
+                string(REGEX REPLACE "lib" "" item ${_name})
+                list(APPEND _lnk "'${item}'")
             endif ()
-
-            list(APPEND _lnk "'${item}'")
         endforeach ()
 
         # defines are a bit more work, because setup.py expects them as tuples
@@ -673,6 +676,7 @@ function(add_setup_py target template)
         string(REGEX REPLACE ";" "," def "${_def}")
         string(REGEX REPLACE ";" "," opt "${_opt}")
         string(REGEX REPLACE ";" "," lnk "${_lnk}")
+        string(REGEX REPLACE ";" "," xlk "${_xlk}")
         string(REGEX REPLACE ";" "," dir "${_dir}")
 
         # TODO: be able to set other name than ext
@@ -682,6 +686,7 @@ function(add_setup_py target template)
                                                 define_macros=[${def}],
                                                 library_dirs=[${dir}],
                                                 libraries=[${lnk}],
+                                                extra_link_args=[${xlk}],
                                                 extra_compile_args=[${opt}])")
 
     endforeach()
