diff --git a/PythonPackage.cmake b/PythonPackage.cmake
index 0f18a3b..9461fd4 100644
--- a/PythonPackage.cmake
+++ b/PythonPackage.cmake
@@ -624,6 +624,8 @@ function(add_setup_py target template)
 
         set(_lnk "")
         foreach (item ${lnk})
+            set(_item ${item})
+            message(STATUS "LNK ITEM: ${item}")
             get_filename_component(_libdir ${item} DIRECTORY)
 
             if (_libdir)
@@ -632,6 +634,13 @@ function(add_setup_py target template)
                 string(REGEX REPLACE "lib" "" item ${item})
             endif ()
 
+            if( ${_item} MATCHES .framework)
+                message(STATUS "::::::::::: FOUND FRAMEWORK ${item}")
+                set(item "-framework ${item}")
+            endif()
+
+            message(STATUS "Appending: ${item}")
+
             list(APPEND _lnk "'${item}'")
         endforeach ()
 
